src/DS/rds.c

```diff
@@ -601,6 +602,13 @@ static void ds_rds_add_xccdf_test_results(xmlDocPtr doc, xmlNodePtr reports,
 {
 	xmlNodePtr root_element = xmlDocGetRootElement(xccdf_result_file_doc);
 
+	if (root_element->ns && root_element->ns->href &&
```

Comment:
Unless you pass an invalid doc it won't be NULL. The rest of the code of the function also assumes it's non-NULL.

What do you suggest I change?